orttraining/orttraining/python/training/ortmodule/_io.py

```diff
@@ -140,13 +162,8 @@ def unflatten(self, flat_args):
         """Unflatten tuple of tensors into args and kwargs"""
 
         args = tuple(flat_args[: self.num_positionals])
-        kwargs = {
-            name: arg
-            for name, arg in zip(
-                self.names[self.num_expanded_positionals_non_none :], flat_args[self.num_positionals :]
-            )
-            if name in self.keyword_names
-        }
+        kwargs = self.kwargs
+        self.kwargs = None
```

Comment:
why we need to store kwargs? the previous implementation generated a kwargs and returned, without storing it

Reply:
recovering nested kwargs (having nested dictionaries) from flattened kwargs is hard. Additionally, we can avoid the complexity of recovering the kwargs if we temporarily store it. This does not make any copies since the tensors inside kwargs are the same as the tensors being passed into the model.